src/sidebar/services/oauth-auth.js

```diff
@@ -167,9 +167,27 @@ function auth($http, $rootScope, $window, OAuthClient,
     if (!tokenInfoPromise) {
       var cfg = serviceConfig(settings);
 
+      if (cfg && cfg.grantToken === 'postMessage') {
```

Comment:
Here I used a special value of the `grantToken` config setting to trigger the postMessage auth behaviour. In the final version we may want to trigger it some other way :man_shrugging: